huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table.go

```diff
@@ -0,0 +1,753 @@
+						},
+						"ordinal": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "The ordinal of a field.",
+						},
+						"table_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The ID of the code table.",
+						},
+						"id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The ID of the field.",
+						},
+					},
+				},
+				Description: "The fields information of the code table.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The description of the code table.",
+			},
+			"directory_path": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The directory path of the code table.",
+			},
+			"created_by": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The user who created the code table.",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The time when the code table was created.",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The time when the code table was updated.",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the code table.",
+			},
+		},
+	}
+}
+
+func resourceArchitectureCodeTableCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var (
+		createCodeTableHttpUrl = "v2/{project_id}/design/code-tables"
+		createCodeTableProduct = "dataarts"
+	)
+	createCodeTableClient, err := cfg.NewServiceClient(createCodeTableProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio client: %s", err)
+	}
+
+	createCodeTablePath := createCodeTableClient.Endpoint + createCodeTableHttpUrl
+	createCodeTablePath = strings.ReplaceAll(createCodeTablePath, "{project_id}", createCodeTableClient.ProjectID)
+
+	createCodeTableOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"workspace": d.Get("workspace_id").(string)},
+	}
+
+	createCodeTableOpt.JSONBody = utils.RemoveNil(buildCreateCodeTableParams(d))
+	createCodeTableResp, err := createCodeTableClient.Request("POST", createCodeTablePath, &createCodeTableOpt)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Architecture code table: %s", err)
+	}
+
+	createCodeTableRespBody, err := utils.FlattenResponse(createCodeTableResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("data.value.id", createCodeTableRespBody)
+	if err != nil || id == nil {
+		return diag.Errorf("error creating DataArts Architecture code table: %s is not found in API response", "id")
+	}
+
+	d.SetId(id.(string))
+
+	yes := hasFieldValues(d)
+	if yes {
```

Comment:
this is more golang style
```suggestion
	if yes := hasFieldValues(d); yes {
```